controllers/main.js

```diff
@@ -69,38 +103,38 @@ module.exports.renderPlay = async (req, res) => {
     sessionLength: playSessionData.length,
     sessionId: id,
     pageNum,
+    userGuessText,
   });
 };
 
-const remapDifficulty = {
-  1: "Very Easy",
-  2: "Easy",
-  3: "Medium",
-  4: "Hard",
-  5: "Very Hard",
+module.exports.renderResults = async (req, res, next) => {
+  const { id } = req.params;
+  const playSessionData = JSON.stringify(
+    await PlaySession.findOne({ sessionId: id })
+  );
+  res.render("main/results", {
+    extraStyles: resultsStyle,
+    playSessionData,
+  });
+  next();
 };
 
-module.exports.renderContinue = async (req, res) => {
-  const user = req.user.username;
-  const lastSessionData = await fetchPlaySessionData(user);
-  const image = await getPixelatedImage(
-    lastSessionData.sessionData[lastSessionData.sessionData.length - 1].imgLink
+module.exports.renderDetailedResults = async (req, res, next) => {
+  const { id } = req.params;
+  const playSessionData = JSON.stringify(
+    await PlaySession.findOne({ sessionId: id })
   );
-  const sessionData = {
-    difficulty: remapDifficulty[lastSessionData.difficulty],
-    length: lastSessionData.length,
-    progress: lastSessionData.sessionData.length,
-    id: lastSessionData.sessionId,
-    image,
-  };
-
-  res.render("main/continue.ejs", { extraStyles: cardStyle, sessionData });
+  res.render("main/detailed-results", {
+    extraStyles: detailedResultsStyle,
+    playSessionData,
+  });
+  next();
 };
 
 module.exports.sendPlayData = async (req, res, next) => {
   const id = uuidv4();
   const playSession = new PlaySession({
-    userId: req.user?.username || "guest",
+    userId: req.user?._id || "guest",
```

Comment:
How is it going to work later on? No criticism or anything just curious as I imagine that the "guest" user will have a lot of play session entries under himself.